accelerator/core/core.c

```diff
@@ -542,7 +543,8 @@ status_t ta_update_node_connection(ta_config_t* const ta_conf, iota_client_servi
   status_t ret = SC_OK;
   for (int i = 0; i < MAX_NODE_LIST_ELEMENTS && ta_conf->iota_host_list[i]; i++) {
     // update new IOTA full node
-    strncpy(service->http.host, ta_conf->iota_host_list[i], HOST_MAX_LEN);
+    strncpy(service->http.host, ta_conf->iota_host_list[i], HOST_MAX_LEN - 1);
```

Comment:
The misuse of `strncpy` might cause potential security issues. I am thinking of introducing wrappers which manipulate strings safely. @marktwtn, how do you think?

Reply:
Sounds good.

Reply:
We might borrow some implementation from Intel safestringlib: https://github.com/intel/safestringlib
Some of the routines in `safestringlib` can be reused (simply copied what we need). It would be regarded as further enhancements.